docs/tracing.md

```diff
@@ -0,0 +1,104 @@
+# Tracing Sorbet's Execution
+
+Sorbet ships with the ability to emit a trace of its execution. This is the most
+lightweight way to
+
+- diagnose why Sorbet is slow on a particular codebase
+- identify areas of Sorbet that could be made faster
+
+This doc will walk through the process of how to collect such information, as
+well as give some rough tips for how to interpret this information.
+
+> **Note**: Making sense of the output of Sorbet's trace information involves
+> general familiarity with the architecture of Sorbet. If you haven't already,
+> you'll want to read [pipeline.md](../pipeline.md), which contains a high-level
+> overview of the various phases of Sorbet.
+
+## Collecting a trace
+
+Sorbet reports trace files using Chrome's [Trace Event Format]. Traces files
+generated using this format can be loaded into `chrome://tracing` and explored
+interactively.
+
+[Trace Event Format]: https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview
+
+To collect a trace, run Sorbet with the `--web-trace-file=<file>` flag:
+
+```bash
+# If you've built Sorbet locally:
+❯ bazel-bin/main/sorbet --web-trace-file=trace.json foo.rb
+
+# or, if you're trying to trace a whole project:
+❯ srb tc --web-trace-file=trace.json
+
+# (if you're a Stripe employee, use `pay exec scripts/bin/typecheck` instead of `srb tc`)
```

Comment:
This will write the trace out to the devbox, is it worth noting that it'll need to be fetched?

Reply:
I might just delete this.

Reply:
You could instead ask them to contact the team in the `#sorbet` channel?